pkg/controllers/termination/terminate.go

```diff
@@ -99,7 +99,7 @@ func (t *Terminator) drain(ctx context.Context, node *v1.Node) error {
 func (t *Terminator) terminate(ctx context.Context, node *v1.Node) error {
 	stored := node.DeepCopy()
 	// Delete the instance associated with node
-	if err := t.CloudProvider.Delete(ctx, machineutil.NewFromNode(node)); err != nil {
+	if err := t.CloudProvider.Delete(ctx, machineutil.NewFromNode(node)); cloudprovider.IgnoreMachineNotFoundError(err) != nil {
```

Comment:
What if we just made Delete idempotent instead of returning NotFound?

Reply:
I think there may be some utility in knowing in the `Delete` whether the object was actually deleted or whether it never existed in the first case. Similar to K8s, this gives more flexibility to the caller to determine how they want to handle that case.

Reply:
I also like how this has parity with K8s errors, where you will get a NotFound error on a Delete on a non-existent object